### PR TITLE
Edits to new Function: `create_ini_settings` in README

### PR DESCRIPTION
**Pull request by jbondpdx**
Also added language highlighting throughout.

### CODE REVIEW

File: README.markdown [hunk 1]
```diff
@@ -146,12 +144,120 @@ glance_api_config { 'HEADER/important_config':
...
+
+For the profile `example`:
+
+~~~puppet
+class profile::example (
+  $settings,
+) {
+    validate_hash($settings)
+    $defaults = { 'path' => '/tmp/foo.ini' }
+    create_ini_settings($settings, $defaults)
```
Comment (hunner) [inline README.markdown]: These three lines should be 2 space indent

File: README.markdown [hunk 2]
```diff
@@ -257,142 +356,40 @@ Specifies whether the subsetting should be present. Valid options: 'present' and
...
+*Optional.* Accepts a hash to be used as the values for any attributes not defined in the first argument.
+ 
+~~~puppet
+`$example = {
+  'section1' => {
+    'setting1' => {
+      'value' => 'value1', 'path' => '/tmp/foo.ini' 
+      } 
+    } 
+  }`
```
Comment (hunner) [inline README.markdown]: These three close curly braces should be UNindented 2 spaces (so they match the indentation of the lines of the corresponding open curlies).

Comment (hunner) [inline README.markdown]: Also, no backticks around this section needed because it's already inside of the triple-tilde

File: README.markdown [hunk 3]
```diff
@@ -257,142 +356,40 @@ Specifies whether the subsetting should be present. Valid options: 'present' and
...
-~~~
-ini_setting { '[section1] setting1':
-    ensure  => present,
-    section => 'section1',
-    setting => 'setting1',
-    value   => 'value1',
-    path    => '/tmp/foo.ini'
-}
-~~~
+##### `defaults`
```
Comment (hunner) [inline README.markdown]: This should be "Second argument: `settings`" to match the previous point

File: README.markdown [hunk 4]
```diff
@@ -67,7 +66,7 @@ ini_subsetting {'sample subsetting':
 
 ###Use a non-standard section header
 
-~~~
+~~~puppet
 default:
    minage = 1
    maxage = 13
```
Comment (hunner) [inline README.markdown]: This section is unclear; the first three lines are the resulting output of the following resource and should be broken into two code blocks, I think... maxage is no where to be seen, so should be removed from the output.

File: README.markdown [hunk 5]
```diff
@@ -51,7 +50,7 @@ The inifile module tries hard not to manipulate your file any more than it needs
 
 Use the `ini_subsetting` type:
 
-~~~
+~~~puppet
 JAVA_ARGS="-Xmx192m -XX:+HeapDumpOnOutOfMemoryError -XX:HeapDumpPath=/var/log/pe-puppetdb/puppetdb-oom.hprof "
```
Comment (hunner) [inline README.markdown]: This should be broken into two code blocks also... the example is configuring the `-Xmx192m` section to be `-Xmx512m`, but then this line should have 512 not 192.

Also there is an extra space before the "

File: README.markdown (hunk 2 above)
Comment (jbondpdx) [inline README.markdown]: Ha, I looked right past the backticks! And thank you. I'll fix it now.

**Pull request merged**